File: offer21-diao-zheng-shu-zu-shun-xu-shi-qi-shu-wei-yu-ou-shu-qian-mian-lcof/python/Solution.py
class Solution:
    def exchange(self, nums: List[int]) -> List[int]:
        i = 0
        j = len(nums) -1
        while i < j:
            if nums[i] % 2 == 1:
                i += 1
                continue
            if nums[j] % 2 == 0:
                j -= 1
                continue
            tmp = nums[i]
            nums[i] = nums[j]
            nums[j] = tmp
            i += 1
            j -= 1
        return nums

File: offer21-diao-zheng-shu-zu-shun-xu-shi-qi-shu-wei-yu-ou-shu-qian-mian-lcof/javascript/Solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var exchange = function(nums) {
    let i = 0
    let j = nums.length - 1
    while (i < j) {
        if (nums[i] % 2 === 1) {
            i++
            continue
        }
        if (nums[j] % 2 === 0) {
            j--
            continue
        }
        [nums[i], nums[j]] = [nums[j], nums[i]]
        i++
        j--
    }
    return nums
};